components/script/dom/document.rs

```diff
@@ -2119,56 +2122,109 @@ impl Document {
             can_gc,
         );
 
-        // If the target has changed then store the current mouse over target for next frame.
-        if target_has_changed {
-            prev_mouse_over_target.set(Some(&new_target));
-        }
+        self.update_current_hover_target_and_status(Some(new_target));
     }
 
-    pub(crate) fn set_cursor(&self, cursor: Cursor) {
-        self.send_to_embedder(EmbedderMsg::SetCursor(self.webview_id(), cursor));
+    fn update_current_hover_target_and_status(&self, new_hover_target: Option<DomRoot<Element>>) {
+        let current_hover_target = self.event_handler.current_hover_target.get();
+        if current_hover_target == new_hover_target {
+            return;
+        }
+
+        let previous_hover_target = self.event_handler.current_hover_target.get();
+        self.event_handler
+            .current_hover_target
+            .set(new_hover_target.as_deref());
+
+        // If the new hover target is an anchor with a status value, inform the embedder
+        // of the new value.
+        let window = self.window();
+        if let Some(target) = self.event_handler.current_hover_target.get() {
+            if let Some(anchor) = target
+                .upcast::<Node>()
+                .inclusive_ancestors(ShadowIncluding::No)
+                .filter_map(DomRoot::downcast::<HTMLAnchorElement>)
+                .next()
+            {
+                let status = anchor
+                    .upcast::<Element>()
+                    .get_attribute(&ns!(), &local_name!("href"))
+                    .and_then(|href| {
+                        let value = href.value();
+                        let url = self.url();
+                        url.join(&value).map(|url| url.to_string()).ok()
+                    });
+                window.send_to_embedder(EmbedderMsg::Status(self.webview_id(), status));
+                return;
+            }
+        }
+
+        // No state was set above, which means that the new value of the status in the embedder
+        // should be `None`. Set that now. If `previous_hover_target` is `None` that means this
```

Comment:
`previous_hover_target` shouldn't be None because of the early return if `current_hover_target == new_hover_target`.